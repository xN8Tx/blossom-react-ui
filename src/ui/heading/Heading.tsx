import React from 'react';

import type { HeadingPropsType } from './Heading.types';

import style from './Heading.module.scss';

export const Heading = ({
  children,
  size,
  weight = 'bold',
  color = 'primary',
  ...props
}: HeadingPropsType) => {
  const className = `${style.Heading} ${style[size]} ${style[weight]} ${style[color]}`;

  const allProps = { className, children, ...props };

  if (size === 'xl' || size === 'l') return <h1 {...allProps} />;
  if (size === 'm') return <h2 {...allProps} />;
  if (size === 's') return <h3 {...allProps} />;
  if (size === 'xs') return <h3 {...allProps} />;
};
