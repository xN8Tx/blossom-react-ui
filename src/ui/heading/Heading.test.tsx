import React from 'react';

import { cleanup, render, screen } from '@testing-library/react';

import { Heading } from './Heading';

describe('Heading', () => {
  afterEach(() => cleanup());
  test('Should render', () => {
    render(
      <Heading color='primary' size='xs' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toBeInTheDocument();
  });

  it('Snapshot XS', () => {
    render(
      <Heading color='primary' size='xs' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toMatchSnapshot();
  });

  it('Snapshot S', () => {
    render(
      <Heading color='primary' size='s' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toMatchSnapshot();
  });

  it('Snapshot M', () => {
    render(
      <Heading color='primary' size='m' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toMatchSnapshot();
  });

  it('Snapshot L', () => {
    render(
      <Heading color='primary' size='l' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toMatchSnapshot();
  });

  it('Snapshot XL', () => {
    render(
      <Heading color='primary' size='xl' weight='bold'>
        Hello world!
      </Heading>,
    );

    const heading = screen.getByText('Hello world!');

    expect(heading).toMatchSnapshot();
  });
});
