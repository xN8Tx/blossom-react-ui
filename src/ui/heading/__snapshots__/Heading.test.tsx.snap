// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Snapshot L 1`] = `
<h1
  class="Heading l bold primary"
>
  Hello world!
</h1>
`;

exports[`Heading Snapshot M 1`] = `
<h2
  class="Heading m bold primary"
>
  Hello world!
</h2>
`;

exports[`Heading Snapshot S 1`] = `
<h3
  class="Heading s bold primary"
>
  Hello world!
</h3>
`;

exports[`Heading Snapshot XL 1`] = `
<h1
  class="Heading xl bold primary"
>
  Hello world!
</h1>
`;

exports[`Heading Snapshot XS 1`] = `
<h3
  class="Heading xs bold primary"
>
  Hello world!
</h3>
`;
